File: Longest Consectuive Sequence/step3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)

        longest_consecutive_sequence = 1

        for num in nums:
            if num - 1 not in num_set:
                length = 1
                while num + length in num_set:
                    length += 1
                longest_consecutive_sequence = max(length, longest_consecutive_sequence)

        return longest_consecutive_sequence

File: Longest Consectuive Sequence/step3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> num_set(nums.rbegin(), nums.rend());
        int longest_consecutive_sequence = 0;

        for ( auto &n: nums) {
            if (!num_set.contains(n-1)) {
                int length = 1;
                while (num_set.contains(n + length)) {
                    length += 1;
                }
                longest_consecutive_sequence = max(length, longest_consecutive_sequence);
            }
        }
        return longest_consecutive_sequence;
    }
};
